site/components/pages/Pledge/lib/pledgeResolver.ts

```diff
@@ -0,0 +1,28 @@
+import { Pledge, PledgeFormValues } from "../types";
+import { DEFAULT_NONCE } from ".";
+
+const DEFAULT_VALUES = {
+  id: "",
+  ownerAddress: "",
+  name: "",
+  nonce: DEFAULT_NONCE,
+  description: "",
+  methodology: "",
+  footprint: 0,
+};
+
+export const pledgeResolver = (pledge: Pledge | null): PledgeFormValues => {
+  if (!pledge) return DEFAULT_VALUES;
+
+  const footprint = pledge.footprint.at(-1)?.total as number;
```

Comment:
bruh how have I never seen this.... i've been typing `array[array.length - 1]` my whole life...

on this line tho again, I'd prefer ugly type guard boilerplate to type assertions.

````
const footprint = pledge.footprint.at(-1);
if (!footprint) {
 // handle
}
```